<?xml version="1.0"?>
<launch>
  <param name="/use_sim_time" value="true"/>

  <!-- Path to MH_03 bag -->
  <!--<arg name="mh_03_bag" default="$(find msckf_mono)/euroc/MH_03_medium.bag"/>-->
  <arg name="mh_03_bag" default="/home/cg/dev_sdb/datasets/ETH_ASL_Datasets/EuRoC_MAV/MH_03_medium.bag"/>

  <!-- Path to MH_03 kalibr calibration -->
  <arg name="kalibr_yaml" default="$(find msckf_mono)/euroc/MH_03_kalibr.yaml"/>

  <group ns="msckf">
    <rosparam command="load" file="$(arg kalibr_yaml)"/>
    <param name="kalibr_camera_name" value="cam0"/>

    <param name="feature_cov" value="2"/>

    <param name="imu_vars/w_var" value="1e-4"/>
    <param name="imu_vars/dbg_var" value="3.6733e-5"/>
    <param name="imu_vars/a_var" value="1e-2"/>
    <param name="imu_vars/dba_var" value="7e-2"/>

    <param name="imu_covars/p_init" value="1e-12"/>
    <param name="imu_covars/q_init" value="1e-5"/>
    <param name="imu_covars/v_init" value="1e-2"/>
    <param name="imu_covars/ba_init" value="1e-2"/>
    <param name="imu_covars/bg_init" value="1e-2"/>

    <param name="max_gn_cost_norm" value="7"/>
    <param name="translation_threshold" value="0.1"/>

    <param name="keyframe_transl_dist" value="0.5"/>
    <param name="keyframe_rot_dist" value="0.5"/>

    <param name="min_track_length" value="5"/>
    <param name="max_track_length" value="50"/>
    <param name="max_cam_states" value="30"/>

    <param name="ransac_threshold" value="0.00000002"/>

    <param name="n_grid_rows" value="10"/>
    <param name="n_grid_cols" value="10"/>

    <node pkg="msckf_mono" name="msckf_mono_node" type="msckf_mono_node" output="screen">
      <remap from="imu" to="/imu0"/>
      <remap from="image_mono" to="/cam0/image_raw"/>
    </node>
  </group>

  <node pkg="rosbag" type="play" name="playback" args="--clock -s 10.0 $(arg mh_03_bag)" output="screen"/>

  <node pkg="rviz" name="msckf_rviz" type="rviz" args="-d $(find msckf_mono)/euroc/rviz.rviz"/>

</launch>
